src/main/java/essenmakanan/recipe/RecipeList.java

```diff
@@ -32,4 +35,38 @@ public Recipe getRecipe(int index) {
         return recipes.get(index);
     }
 
+    public void viewSpecificRecipe(int index) {
+        Ui.drawDivider();
+        if (index < 0 | index >= recipes.size()) {
+            System.out.println("We have " + recipes.size() + "recipes right now and the given input is invalid.");
+            return;
+        }
+        Recipe recipe = recipes.get(index-1);
+        listAllSteps(recipe);
+    }
+
+    public void viewSpecificRecipe(String title) {
+        Ui.drawDivider();
+        Recipe recipe = recipes.stream().filter(recipe1 -> recipe1.getTitle().equals(title)).findFirst().orElse(null);
```

Comment:
can add a new line for each function called. 
![image](https://github.com/AY2324S1-CS2113-F11-2/tp/assets/72329348/c491c105-7660-41e4-9e5a-89f1799606b4)

and can change recipe1 name to filteredRecipe

Reply:
Modified, it's more readable now :D